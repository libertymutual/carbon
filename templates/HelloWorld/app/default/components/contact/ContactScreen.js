import React from 'react';
import PropTypes from 'prop-types';
import { Button, ScrollView, StyleSheet, Text } from 'react-native';
import { connect } from 'react-redux';

import * as PresentationConstants from '../../constants/PresentationConstants';

/**
 * Contact Screen
 * @extends {React.Component}
 */
class ContactScreen extends React.Component {

  /**
   * Home Screen Constructor
   * @constructor
   * @param {Object} ...args Arguments passed to the constructor
   */
  constructor(...args) {
    super(...args);

    this._navigateBack = this._navigateBack.bind(this);
  }

  /**
   * Renders the components
   * @override
   * @returns {JSX}
   */
  render() {
    return (<ScrollView>
      <Text style={styles.contactScreenHeader}>{PresentationConstants.CONTACT_WELCOME}</Text>
      <Button
        onPress={this._navigateBack}
        title={PresentationConstants.GO_BACK}
      />
    </ScrollView>);
  }

  /**
   * Navigates back to the previous screen
   * @private
   */
  _navigateBack = () => {
    this.props.navigation.goBack();
  }
}

/**
 * Navigation Options
 */
ContactScreen.navigationOptions = {
  title: PresentationConstants.PAGE_TITLE_CONTACT_SCREEN,
};

/**
 * Prop Types
 */
ContactScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

/**
 * Maps State To Props
 * @returns {Object) Object representing the parts of state
 * that the component should subscribe to
 */
function mapStateToProps() {
  return {};
}

/**
 * Maps Dispatch Actions To Props
 * @returns {Object) Object representing the actions
 * that should be exposed to the component via props
 */
function mapDispatchToProps() {
  return {};
}

/**
 * Default export wrapped in a connect to expose state & actions via props
 */
export default connect(mapStateToProps, mapDispatchToProps)(ContactScreen);

/**
 * Component Styles
 */
const styles = StyleSheet.create({
  contactScreenHeader: {
    margin: 14,
    fontWeight: 'bold',
  },
});
