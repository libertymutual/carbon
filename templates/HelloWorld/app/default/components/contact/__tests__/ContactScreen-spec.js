import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ContactScreen from '../ContactScreen';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
let store;

describe('ContactScreen tests', () => {
  beforeEach(() => {
    // Sets up a new store
    store = mockStore({});
  });

  test('Contact Screen renders correctly', () => {
    const tree = renderer.create(
      <ContactScreen store={store} navigation={{}}/>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
