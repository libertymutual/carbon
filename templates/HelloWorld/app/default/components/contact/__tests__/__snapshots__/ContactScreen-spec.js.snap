// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactScreen tests Contact Screen renders correctly 1`] = `
<RCTScrollView>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "margin": 14,
        }
      }
    >
      Welcome To The Default Feature Contact Us Screen
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
